File: frontend/src/types/actaNota.ts
import type { AsignacionDocente } from './asignacionDocente'
import type { Usuario } from './usuario'

export type EstadoActa = 'ABIERTA' | 'OBSERVADA' | 'VALIDADA' | 'CERRADA'

export interface ActaNota {
  id: number
  asignacion: AsignacionDocente
  estado: EstadoActa
  fechaCierre?: string | null
  fechaValidacion?: string | null
  validadaPor?: Usuario | null
  observacion?: string | null
}

export const ESTADO_ACTA_LABELS: Record<EstadoActa, string> = {
  ABIERTA: 'Abierta',
  OBSERVADA: 'Observada',
  VALIDADA: 'Validada',
  CERRADA: 'Cerrada',
}

File: sistema-academico/src/main/java/com/universidad/sistema_academico/entity/ActaNota.java
package com.universidad.sistema_academico.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "actas_nota")
public class ActaNota {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(optional = false)
    @JoinColumn(name = "asignacion_id", unique = true)
    private AsignacionDocente asignacion;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private EstadoActa estado = EstadoActa.ABIERTA;

    private LocalDateTime fechaCierre;

    private LocalDateTime fechaValidacion;

    @ManyToOne
    @JoinColumn(name = "validada_por_id")
    private Usuario validadaPor;

    @Column(length = 300)
    private String observacion;
}
